### DISCUSSION TITLE
How do the developers make the ISO?

### DISCUSSION THREAD

**Discussion by fogel-j** [Security-Onion-Solutions/securityonion#12414] [General] [answered]
I'm interested in making an ISO similar to SO that will install the base OS image and then run post installation of all necessary software and system compliance scripts. The developers have made this very streamlined so that you only need one ISO file, you boot from it, and you're good to go. How were you able to accomplish this? Is there any resources you can point me to?

I'm working off a Rocky Linux 8 OS.

**Comment by TotieBash** [comment, marked as answer]
Kickstart install method is what you are looking for.  Google linux kickstart and it will take you to what you are looking for.  If you want to reverse engineer the SO kickstart method then I would start by mounting the SO image iso and look at /BOOT/grub.cfg.  Within that grub.cfg file it will reference the varius kickstart file, I would start looking at /ks.cfg.  The ks.cfg kickstart file will hold most of the installation script answer file.   You can study the file and find how SO configures firewall, root pw, selinux, partition, etc, etc. 

Create a bootable USB thumbrive:  I use Unetbootin to create a bootable USB thumbdrive (or Rufus) from a Linux iso image.  For instance I would download RHEL9.3 or Oracle Linux 9.3 iso and would use Rufus or Unetbootin to create bootable USB thumbdrive from the iso.  I use the thumbrive test method because I know I would have to massage ks.cfg file many many times until im happy with it.  Then I would edit the bootable thumbrive:/BOOT/grub.cfg to make my changes to point to my ks.cfg.  Obviously you also copy the ks.cfg in to the root directory.  

Then finally if you really need iso there are tools to convert the bootable USB thumbrive to bootable iso.  I had success using old imgburn software in the past but I honestly dont do it that much.  I use the bootable thumbdrive method for me because it is convenient for me to make edit to kickstart file.  For instance if I need to change static IP or hostname of the machine, I just edit couple of lines on kickstart and im done.
